pkg/graveler/graveler.go

```diff
@@ -1740,10 +1740,13 @@ func (g *KVGraveler) retryBranchUpdate(ctx context.Context, repository *Reposito
 			try += 1
 			return err
 		}
+		if err != nil {
+			return backoff.Permanent(err)
+		}
 		return nil
 	}, bo)
-	if try == setTries {
-		return fmt.Errorf("update branch: %w", ErrTooManyTries)
+	if try > setTries {
+		return backoff.Permanent(fmt.Errorf("update branch: %w", ErrTooManyTries))
```

Comment:
No need to wrap this with backoff.Permanent

Reply:
oops